src/Compilers/VisualBasic/Portable/Binding/Binder.vb

```diff
@@ -452,6 +453,60 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return type.GetUseSiteErrorInfo()
         End Function
 
+        Private Function GetInternalXmlHelperType(syntax As VisualBasicSyntaxNode, diagBag As DiagnosticBag) As NamedTypeSymbol
+            Debug.Assert(Me.ContainingType IsNot Nothing)
+            Dim containingAssembly = Me.ContainingType.ContainingAssembly
```

Comment:
As we discussed offline, that makes a lot of sense and eliminates the need for a bunch of validation after we find it.